### ISSUE TITLE
Cannot open IVRVirtualDisplay::Present::SharedTextureHandle_t with hardware adapter specified

### ISSUE THREAD

**Issue by ShootingKing-AM** [ValveSoftware/openvr#1503]
Cannot open IVRVirtualDisplay::Present::SharedTextureHandle_t with hardware adapter specifed in `D3D11CreateDevice`, like

```c++
IDXGIFactory *pDXGIFactory;
hr = CreateDXGIFactory(__uuidof(IDXGIFactory), (void **)&pDXGIFactory);
if (SUCCEEDED(hr)) {
	int adapter = 0;
	if (FAILED(pDXGIFactory->EnumAdapters(adapter, &pAdapter)))
		pAdapter = NULL;
	pDXGIFactory->Release();
}
		
hr = D3D11CreateDevice(pAdapter, pAdapter ? D3D_DRIVER_TYPE_UNKNOWN: D3D_DRIVER_TYPE_HARDWARE, NULL, creationFlags, NULL, 0, D3D11_SDK_VERSION, &device_hwctx->device, NULL, NULL);

ID3D11Texture2D *inpTex;
hr = avHWDevice->OpenSharedResource((HANDLE)gfxSharedHandle, __uuidof(ID3D11Texture2D), (void **)&inpTex); // Error: Invalid Argument
```

But this works,
```c++
hr = D3D11CreateDevice(NULL, D3D_DRIVER_TYPE_HARDWARE, NULL, creationFlags, NULL, 0, D3D11_SDK_VERSION, &device_hwctx->device, NULL, NULL);

ID3D11Texture2D *inpTex;
hr = avHWDevice->OpenSharedResource((HANDLE)gfxSharedHandle, __uuidof(ID3D11Texture2D), (void **)&inpTex); // Successfully opened
```
Yes, i have two hardware GPU adapters, one Intel and one Nvidia GTX 1050. and in both cases the adapter thats being used is  `10de:1c8d (NVIDIA GeForce GTX 1050)`.

OS: Windows 10
DirectX: 12 (Feature Level: 11 is supported from dxdiag)

**Comment by Rectus** [comment]
Is that ternary supposed to set the driver type to D3D_DRIVER_TYPE_UNKNOWN?

**Comment by ShootingKing-AM** [comment]
> Is that ternary supposed to set the driver type to D3D_DRIVER_TYPE_UNKNOWN?
yup.

**Comment by Rectus** [comment]
This issue looks like it's similar to yours, and they seem to have fixed it by using the DXGI 1.1 version of CreateDXGIFactory.
#https://github.com/ValveSoftware/openvr/issues/539

**Comment by ShootingKing-AM** [comment]
> This issue looks like it's similar to yours, and they seem to have fixed it by using the DXGI 1.1 version of CreateDXGIFactory.
> ##539

Thankyou very much, this fixed the issue 👍
